hooks/terraform_wrapper_module_for_each.sh

```diff
@@ -0,0 +1,424 @@
+
+This wrapper does not implement any extra functionality.
+
+## Usage with Terragrunt
+
+`terragrunt.hcl`:
+
+```hcl
+terraform {
+  source = "tfr:///MODULE_REPO_ORG/MODULE_REPO_SHORTNAME/MODULE_REPO_PROVIDER//WRAPPER_PATH"
+  # Alternative source:
+  # source = "git::git@github.com:MODULE_REPO_ORG/terraform-MODULE_REPO_PROVIDER-MODULE_REPO_SHORTNAME.git?ref=master//WRAPPER_PATH"
+}
+
+inputs = {
+  defaults = { # Default values
+    create = true
+    tags = {
+      Terraform   = "true"
+      Environment = "dev"
+    }
+  }
+
+  items = {
+    my-item = {
+      # omitted... can be any argument supported by the module
+    }
+    my-second-item = {
+      # omitted... can be any argument supported by the module
+    }
+    # omitted...
+  }
+}
+```
+
+## Usage with Terraform
+
+```hcl
+module "wrapper" {
+  source = "MODULE_REPO_ORG/MODULE_REPO_SHORTNAME/MODULE_REPO_PROVIDER//WRAPPER_PATH"
+
+  defaults = { # Default values
+    create = true
+    tags = {
+      Terraform   = "true"
+      Environment = "dev"
+    }
+  }
+
+  items = {
+    my-item = {
+      # omitted... can be any argument supported by the module
+    }
+    my-second-item = {
+      # omitted... can be any argument supported by the module
+    }
+    # omitted...
+  }
+}
+```
+
+## Example: Manage multiple S3 buckets in one Terragrunt layer
+
+`eu-west-1/s3-buckets/terragrunt.hcl`:
+
+```hcl
+terraform {
+  source = "tfr:///terraform-aws-modules/s3-bucket/aws//wrappers"
+  # Alternative source:
+  # source = "git::git@github.com:terraform-aws-modules/terraform-aws-s3-bucket.git?ref=master//wrappers"
+}
+
+inputs = {
+  defaults = {
+    force_destroy = true
+
+    attach_elb_log_delivery_policy        = true
+    attach_lb_log_delivery_policy         = true
+    attach_deny_insecure_transport_policy = true
+    attach_require_latest_tls_policy      = true
+  }
+
+  items = {
+    bucket1 = {
+      bucket = "my-random-bucket-1"
+    }
+    bucket2 = {
+      bucket = "my-random-bucket-2"
+      tags = {
+        Secure = "probably"
+      }
+    }
+  }
+}
+```'
+
+function terraform_module_wrapper_() {
+  local args
+  read -r -a args <<< "$1"
+
+  local root_dir
+  local module_dir="" # values: empty (default), "." (just root module), or a single module (e.g. "modules/iam-user")
+  local wrapper_dir="wrappers"
+  local wrapper_relative_source_path="../" # From "wrappers" to root_dir.
+  local module_repo_org
+  local module_repo_name
+  local module_repo_shortname
+  local module_repo_provider
+  local dry_run="false"
+  local verbose="false"
+
+  root_dir=$(git rev-parse --show-toplevel 2> /dev/null || pwd)
+  module_repo_org="terraform-aws-modules"
+  module_repo_name=${root_dir##*/}
+  module_repo_shortname="${module_repo_name#terraform-aws-}"
+  module_repo_provider="aws"
+
+  for argv in "${args[@]}"; do
+
+    IFS="=" read -r -a onearg <<< "$argv"
+
+    local key="${onearg[0]}"
+    local value="${onearg[1]}"
```

Comment:
What thought I just had was that in case when *value* contains equal sign (e.g. when input to this `for` loop has an element of `this_key=has_value_with=this_sign`), this scaffold will extract only the part of it before first equal sign. I'm not quite sure of whether this can be a matter or a real use case, though to eliminate possible confusion this can be rewritten as below, which requires no `read` and uses Bash builtin expansion:
```suggestion
    IFS="=" read -r -a onearg <<< "$argv"

    local key="${argv%%=*}"
    local value="${argv#*=}"
```